### DISCUSSION TITLE
JPEG XL color profile?

### DISCUSSION THREAD

**Discussion by VcSaJen** [dlemstra/Magick.NET#1048] [Help]
I'm trying to read this file from JPEG XL conformance tests: https://github.com/libjxl/conformance/blob/a5e01103a2c09b85158638de367387d051474b96/testcases/bench_oriented_brg/input.jxl?raw=true . It's supposed to be red bench image. When I open it with Magick.NET, it's a green bench. I tried to fix it by adding `image.SetProfile(ColorProfile.SRGB);` command, but it seemed to do nothing. Also tried `image.TransformColorSpace(ColorProfile.SRGB); image.Strip();`, which also seems to do nothing. `image.GetColorProfile()` returns null.
What should I do to open it correctly, so that it's a red bench?

**Comment by dlemstra** [comment]
I am also getting a great image and I wonder if this was also recently fixed in the jpeg-xl library. I guess we will need to wait for their next release to test this again.

**Reply by dlemstra** [reply]
And I just spoke with one of the jpeg-xl library developers and they told me that not all those files are working yet.

**Reply by VcSaJen** [reply]
What do you mean? libjxl does not support returning color profile?

**Reply by dlemstra** [reply]
It does support that and I just fixed a bug to make sure this will be returned for that image that you linked. But that will not become available in the next release of Magick.NET because the release is already prepared and will be published later today.
